org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/grid/IQuickEntry.java

```diff
@@ -0,0 +1,67 @@
+/******************************************************************************
+ * Copyright (C) Contributors                                             *
+ * Product: iDempiere ERP & CRM Smart Business Solution                       *
+ * This program is free software; you can redistribute it and/or modify it    *
+ * under the terms version 2 of the GNU General Public License as published   *
+ * by the Free Software Foundation. This program is distributed in the hope   *
+ * that it will be useful, but WITHOUT ANY WARRANTY; without even the implied *
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.           *
+ * See the GNU General Public License for more details.                       *
+ * You should have received a copy of the GNU General Public License along    *
+ * with this program; if not, write to the Free Software Foundation, Inc.,    *
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.                     *
+ * Contributors:                                                              *
+ *   Andreas Sumerauer                                                        *
+ *****************************************************************************/
+
+package org.adempiere.webui.grid;
+
+import org.adempiere.webui.ISupportMask;
+import org.adempiere.webui.event.ValueChangeEvent;
+import org.adempiere.webui.event.ValueChangeListener;
+import org.zkoss.zk.ui.Component;
+import org.zkoss.zk.ui.event.Event;
+import org.zkoss.zk.ui.event.EventListener;
+
+/**
+ * Quick Entry Window
+ * Interface based on class WQuickEntry by Carlos Ruiz
+ * @Author Andreas Sumerauer
+ */
+public interface IQuickEntry extends  Component, ISupportMask, EventListener<Event>, ValueChangeListener
```

Comment:
This design change make it necessary to do a window cast at WTableDirEditor. Unless we can find a way to get rid of that class cast, this probably not a good design change. It implies that you can create a IQuickEntry implementation that doesn't extends Window and will cause class cast exception.

Reply:
I am aware of that inconsistency and I don't like it either. 
I should be able to solve it using an abstract base class AbstractWQuickEntry instead of the Interface.
(Should have thought of that earlier)